### PR TITLE
Add keywords to the document overview

### LINKED ISSUE #4370: Schlagworte auf Dokument-Detailansicht nicht ersichtlich
Die Schlagwörter eines Dokumentes sollten, ähnlich wie in der Dossierübersicht auch auf der Dokument-Detailansicht dargestellt werden. 

# Soll
bildschirmfoto_2018-05-28_um_17_53_16

### PR DESCRIPTION
**Pull request by tarnap**
screen shot 2018-05-30 at 08 03 36

screen shot 2018-06-06 at 09 41 40

Resolves #4370

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog
 2018.4.0 (unreleased)
 ---------------------
 
+- Do not display empty parentheses if no abbreviation available. [tarnap]
```
Comment (phgross) [inline docs/HISTORY.txt]: ???

File: opengever/document/browser/overview.py [hunk 2]
```diff
@@ -146,6 +146,7 @@ def get_metadata_config(self):
             CustomRow(self.render_creator_link,
                       label=_('label_creator', default='creator')),
             FieldRow('IDocumentMetadata.description'),
+            FieldRow('IDocumentMetadata.keywords'),
```
Comment (phgross) [inline opengever/document/browser/overview.py]: The screenshot looks like, that the keywords aren't linked as specified in the issue #4370.

File: docs/HISTORY.txt (hunk 1 above)
Comment (tarnap) [reply docs/HISTORY.txt]: Ah!! Here it is! I failed at staging 🙈

File: opengever/document/browser/overview.py (hunk 2 above)
Comment (tarnap) [reply opengever/document/browser/overview.py]: Indeed, I need to find out what's happening there.

Comment (tarnap) [reply opengever/document/browser/overview.py]: What should these be linked to? Is there a view which filters for keywords or such?

Comment (phgross) [reply opengever/document/browser/overview.py]: As already mentioned in the issue #4370, take a look at the dossier implementation.

File: opengever/document/browser/overview.py [hunk 3]
```diff
@@ -258,6 +262,16 @@ def render_checked_out_link(self):
...
         return ''
 
+    def render_keywords(self):
+        linked_keywords = []
+        for keyword in IDocumentMetadata(self.context).keywords:
+            url = u'{}/@@search?Subject={}'.format(
+                api.portal.get().absolute_url(),
+                quote_plus(safe_unicode(keyword).encode('utf-8')))
+            linked_keywords.append(
+                '<a href="{}">{}</a>'.format(url, keyword))
```
Comment (phgross) [inline opengever/document/browser/overview.py]: This will fail with an UnicodeEncodeError, when the keyword contains umlauts.

Comment (phgross) [inline opengever/document/browser/overview.py]: This is not XSS safe.

File: opengever/document/browser/overview.py [hunk 4]
```diff
@@ -258,6 +262,16 @@ def render_checked_out_link(self):
...
+        return ', '.join(linked_keywords)
```
Comment (phgross) [inline opengever/document/browser/overview.py]: Same here: potential encoding problem

File: opengever/document/browser/overview.py [hunk 5]
```diff
@@ -146,6 +148,8 @@ def get_metadata_config(self):
             CustomRow(self.render_creator_link,
                       label=_('label_creator', default='creator')),
             FieldRow('IDocumentMetadata.description'),
+            CustomRow(self.render_keywords,
```
Comment (phgross) [inline opengever/document/browser/overview.py]: I think it would be easier to use a TemplateRow and let TAL handle the "encoding work"

Comment (tarnap) [reply opengever/document/browser/overview.py]: I was expecting something like this.
I reimplemented the row with a `TemplateRow`

**Comment by tarnap** [comment]
Changes were made, please re-re-review this pull-request.

**Comment by tarnap** [comment]
@deiferni: A test was added. Please re-re-re-review :)

**Pull request merged**